src/V3SchedTiming.cpp

```diff
@@ -384,7 +384,7 @@ void transformForks(AstNetlist* const netlistp) {
                 }
                 // Put the begin's statements in the function, delete the begin
                 newfuncp->addStmtsp(nodep->stmtsp()->unlinkFrBackWithNext());
-                if (m_beginNeedProcess) {
+                if (m_beginNeedProcess || m_funcp->needProcess()) {
```

Comment:
What if `!m_beginNeedProcess && !m_beginHasAwaits`, but `m_funcp->needProcess()`? We won't enter the outer if. Is that ok?